File: projeto_de_bloco/TPs/tp6/tp6/ViewController.m
//
//  ViewController.m
//  tp6
//
//  Created by Magno Valdetaro on 4/9/16.
//  Copyright © 2016 infnet. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
{
    NSMutableArray *listaCidades;
    
    //NSArray *listaCidades;
    
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    listaCidades = [[NSMutableArray alloc] initWithArray:[NSArray arrayWithObjects:@"Rio de Janeiro", @"São Paulo", @"Belo Horizonte", @"Cabo Frio", @"Campinas", @"Londres", @"Paris", @"Bangkok", @"Cingapura", @"Hong Kong", @"Madri",nil]];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return listaCidades.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *cellIdentifier = @"TabelaSimples";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
    }
    cell.textLabel.text = listaCidades[indexPath.row];
    return cell;
}

- (IBAction)btnAdd:(id)sender {
    
    NSString *novaCidade = self.inNovaCidade.text;
    [listaCidades insertObject: novaCidade atIndex:0];
    //[listaCidades addObject: novaCidade];
    [self.tableView reloadData];
    
    
    //NSUInteger tamanhoLista = listaCidades.count;
    //NSLog(@"%@", novaCidade);
    //NSLog(@"%lu", tamanhoLista);
}
@end

File: projeto_de_bloco/pratica/TuristandoRio/TuristandoRio/ViewController.swift
//
//  ViewController.swift
//  TuristandoRio
//
//  Created by Magno Valdetaro on 5/14/16.
//  Copyright © 2016 infnet. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
    
    @IBOutlet weak var tableView: UITableView!
    var places:NSArray?
    
    

    override func viewDidLoad() {
        super.viewDidLoad()
        
        if let bundlePath = NSBundle.mainBundle().pathForResource("Places", ofType: "plist"){
            if let dicionario = NSDictionary(contentsOfFile: bundlePath){
                places = dicionario["places"] as? NSArray
                tableView.reloadData()
                
            }
        }
        
        
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return (places?.count)!
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        
        let cell:PlaceTableViewCell = tableView.dequeueReusableCellWithIdentifier("placeCell", forIndexPath: indexPath) as! PlaceTableViewCell
        
        if let place:NSDictionary = places![indexPath.row] as? NSDictionary {
            cell.lbNome.text = place["name"] as? String
            cell.lbAddress.text = place["address"] as? String
            
            // crop image
            let urlImage = UIImage(named: place["image"] as! String)
            let image: UIImage = ImageUtil().cropToSquare(image: urlImage!)
            
            cell.imageCell.image = image
        }
        return cell
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if let index = tableView.indexPathForSelectedRow {
            let detailVC = segue.destinationViewController as! DetailViewController
            let dictionary = places![index.row] as? NSDictionary
            detailVC.dicPlace = dictionary
        }
    }
    

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

